qemu/tests/nvdimm.py

```diff
@@ -107,50 +108,68 @@ def run(test, params, env):
+                test.error("Creating dax failed!")
         if not os.path.exists(params["nv_backend"]):
-            raise error.TestError("Check nv_backend in host failed!")
+            test.error("Check nv_backend in host failed!")
         params["start_vm"] = "yes"
         vm_name = params['main_vm']
         env_process.preprocess_vm(test, params, env, vm_name)
 
-    nvdimm_test = NvdimmTest(params, env)
+    nvdimm_test = NvdimmTest(test, params, env)
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
     try:
-        error.context("Login to the guest", logging.info)
+        error_context.context("Login to the guest", logging.info)
         login_timeout = int(params.get("login_timeout", 360))
         nvdimm_test.session = vm.wait_for_login(timeout=login_timeout)
-        error.context("Verify nvdimm in monitor and guest", logging.info)
+        error_context.context("Verify nvdimm in monitor and guest", logging.info)
         nvdimm_test.verify_nvdimm(vm)
-        error.context("Format and mount nvdimm in guest", logging.info)
+        error_context.context("Format and mount nvdimm in guest", logging.info)
         nvdimm_test.mount_nvdimm()
         nv_file = params.get("nv_file", "/mnt/nv")
-        error.context("Create a file in nvdimm mount dir in guest, and get "
-                      "original md5 of the file", logging.info)
+        error_context.context("Create a file in nvdimm mount dir in guest, and get "
+                              "original md5 of the file", logging.info)
         dd_cmd = "dd if=/dev/urandom of=%s bs=1K count=200" % nv_file
         nvdimm_test.run_guest_cmd(dd_cmd)
         orig_md5 = nvdimm_test.md5_hash(nv_file)
         nvdimm_test.umount_nvdimm()
         nvdimm_test.session = vm.reboot()
-        error.context("Verify nvdimm after reboot", logging.info)
+        error_context.context("Verify nvdimm after reboot", logging.info)
         nvdimm_test.verify_nvdimm(vm)
         nvdimm_test.mount_nvdimm(format_device="no")
         new_md5 = nvdimm_test.md5_hash(nv_file)
-        error.context("Compare current md5 to original md5", logging.info)
+        error_context.context("Compare current md5 to original md5", logging.info)
         if new_md5 != orig_md5:
-            raise error.TestFail("'%s' changed. The original md5 is '%s',"
-                                 " current md5 is '%s'"
-                                 % (nv_file, orig_md5, new_md5))
+            test.fail("'%s' changed. The original md5 is '%s', current md5 is '%s'"
+                      % (nv_file, orig_md5, new_md5))
         nvdimm_test.umount_nvdimm()
-        error.context("Check if error and calltrace in guest", logging.info)
+        error_context.context("Check if error and calltrace in guest", logging.info)
         vm.verify_kernel_crash()
 
     finally:
+        if nvdimm_dax:
+            try:
+                process.system(params["del_dax_cmd"], timeout=login_timeout, shell=True)
```

Comment:
I prefer to use one timeout value, not "login_timeout".  As it has nothing to do with login.

Reply:
Change all login_timeout to another name "nvdimm_timeout", Thanks!

Reply:
@ybduan  this is not what i mean,  login_timeout is ok , as it is used for wait_for_login and so on.
you can use 240 instead. what do you think?

Reply:
@ybduan  what do think about the latest comment ?

Reply:
Modified, Thanks!